lib/perl/Genome/Model/CwlPipeline/Command/Run.pm

```diff
@@ -177,6 +179,97 @@ sub run_cromwell {
     $self->_stage_cromwell_outputs($results_dir);
 }
 
+sub run_cromwell_gcp {
+    my $self = shift;
+    my $yaml = shift;
+    my $results_dir = shift;
+
+    my $zip_deps = Genome::Config::get('workflow_deps_zip');
+    my $workflow_options = Genome::Config::get('cromwell_workflow_options');
+    my $bucket = Genome::Config::get('cromwell_gcp_bucket');
+
+    my $logdir = $self->build->log_directory;
+    my $cromwell_url = Genome::Cromwell->server_url;
+    my $lsb_sub_additional = Genome::Config::get('lsb_sub_additional');
+    my $queue = Genome::Config::get('lsf_queue_build_worker');
+    my $user_group = Genome::Config::get('lsf_user_group');
+    my $main_workflow_file = $self->build->model->main_workflow_file;
+
+    my $poll_interval_seconds = 30;
+
+    # Cloudize workflow
+    my $cloud_yaml = $yaml =~ s/.ya?ml/_cloud.yaml/r;
+    my $guard = Genome::Config::set_env('lsb_sub_additional', 'docker(jackmaruska/cloudize-workflow:latest)');
+    delete local $ENV{BOTO_CONFIG};
+
+    Genome::Sys::LSF::bsub::bsub(
+        queue => $queue,
+        user_group => $user_group,
+        resource_string => 'rusage[mem=512M:internet2_upload_mbps=500]',
+        wait_for_completion => 1,
+        log_file => "$logdir/cloudize_workflow.log",
+        cmd => [
+            "python3",
+            "/opt/scripts/cloudize-workflow.py",
+            $bucket,
+            $main_workflow_file,
+            $yaml,
+            "--output=$cloud_yaml"] );
+
+    print "input yaml $yaml\ncloud yaml $cloud_yaml\n";
+    my $workflow_id = Genome::Cromwell->submit_workflow(
+        $main_workflow_file, $cloud_yaml, $zip_deps, $workflow_options )->{id};
+
+    print "Submitted workflow id=" . $workflow_id . "\n";
+
+    # Poll until done
+    my $status;
+    do {
+        sleep $poll_interval_seconds;
+        $status = Genome::Cromwell->status($workflow_id)->{status};
+    } while ($status eq "Running" || $status eq "Submitted");
+
+    # Pull Outputs
+    if ($status eq "Succeeded") {
+        Genome::Sys::LSF::bsub::bsub(
+            queue => $queue,
+            user_group => $user_group,
+            resource_string => 'rusage[mem=512M:internet2_download_mbps=500]',
+            wait_for_completion => 1,
+            log_file => "$logdir/pull_outputs.log",
+            cmd => [
+                "python3", "/opt/scripts/pull_outputs.py",
+                $workflow_id, "--output=$results_dir", "--cromwell-url=$cromwell_url"
+            ]
+            );
+        $self->_generate_timing_report($workflow_id);
+    } else {
+        $self->fatal_message("Workflow $workflow_id has non-succeeded status $status using workflow definition $main_workflow_file. Logs at $logdir");
+    }
+
+    $self->_fetch_cromwell_log($workflow_id, $workflow_options, $logdir);
+
+    $guard = Genome::Config::set_env('lsb_sub_additional', $lsb_sub_additional);
```

Comment:
`Genome::Config::set_env` is a `Scope::Guard` object that resets the value when the reference goes out of scope, so this particular line doesn't do too much.  The old value was about to be restored anyway.  If you do want to be explicit about resetting the previous value, this could also be `undef $guard` instead.

Reply:
I didn't know that it automatically reset when it goes out of scope, neat